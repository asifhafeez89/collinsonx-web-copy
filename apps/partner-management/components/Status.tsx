import { Box, Text } from '@collinsonx/design-system/core';
import {
  Confirmed,
  Pending,
  Declined,
} from '@collinsonx/design-system/assets/icons';

export type type = 'warning' | 'success' | 'danger';
export type typeProps = {
  type: type;
  children: JSX.Element | string;
};

const styles = {
  padding: '8px 16px',
  borderRadius: 4,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  gap: 10,
  height: 44,
  fontSize: 14,
  fontWeight: 500,
  color: '#000000',
};

interface Props {
  children: JSX.Element | string;
}

const statusMap = {
  warning: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#FEF3DA',
        border: '2px solid #FAB005',
      }}
    >
      <Pending />
      <Text>{children}</Text>
    </Box>
  ),
  success: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#EEF9E7',
        border: '2px solid #54C50D',
      }}
    >
      <Confirmed />
      <Text>{children}</Text>
    </Box>
  ),
  danger: ({ children }: Props) => (
    <Box
      sx={{
        ...styles,
        background: '#FDECEC',
        border: '2px solid #F03E3E',
      }}
    >
      <Declined />
      <Text>{children}</Text>
    </Box>
  ),
};

export default function type({ type, children }: typeProps) {
  const Component = statusMap[type];
  return <Component>{children}</Component>;
}
