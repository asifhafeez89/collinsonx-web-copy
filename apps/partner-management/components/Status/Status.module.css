.base {
  padding: 8px 16px;
  border-radius: 4;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 10;
  height: 44;
  font-size: 14;
  font-weight: 500;
  color: #000000;
}
.statusPending {
  background: #fef3da;
  border: 2px solid #fab005;
}
.statusConfirmed {
  background: #eef9e7;
  border: 2px solid #54c50d;
}
.statusDeclined {
  background: #fdecec;
  border: 2px solid #f03e3e;
}
