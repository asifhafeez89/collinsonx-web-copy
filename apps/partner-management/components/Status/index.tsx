import { Box, Text } from '@collinsonx/design-system/core';
import cx from 'clsx';
import {
  Confirmed,
  Pending,
  Declined,
} from '@collinsonx/design-system/assets/icons';

import classes from './Status.module.css';

export type type = 'warning' | 'success' | 'danger';
export type typeProps = {
  type: type;
  children: JSX.Element | string;
};

interface Props {
  children: JSX.Element | string;
}

const statusMap = {
  warning: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusPending])}>
      <Pending />
      <Text>{children}</Text>
    </Box>
  ),
  success: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusConfirmed])}>
      <Confirmed />
      <Text>{children}</Text>
    </Box>
  ),
  danger: ({ children }: Props) => (
    <Box className={cx([classes.base, classes.statusDeclined])}>
      <Declined />
      <Text>{children}</Text>
    </Box>
  ),
};

export default function type({ type, children }: typeProps) {
  const Component = statusMap[type];
  return <Component>{children}</Component>;
}
