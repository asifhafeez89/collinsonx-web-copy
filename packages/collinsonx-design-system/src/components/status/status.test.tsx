import renderer from 'react-test-renderer';
import  Status  from '.';

describe('<Status />', () => {
    it('renders Status', () => {
        const tree = renderer
            .create(
                <Status
                    status="PENDING"
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})