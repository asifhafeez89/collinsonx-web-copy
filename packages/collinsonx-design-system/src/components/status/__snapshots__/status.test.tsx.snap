// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> renders Status 1`] = `
<div
  className="mantine-Text-root mantine-1xk0dk8"
>
  Booking pending
</div>
`;
